bootstrap/internal/controller/infraenv_controller.go

```diff
@@ -18,23 +18,40 @@ package controller
 
 import (
 	"context"
+	"fmt"
+	"net/url"
+	"os"
 
 	bootstrapv1alpha1 "github.com/openshift-assisted/cluster-api-agent/bootstrap/api/v1alpha1"
 	logutil "github.com/openshift-assisted/cluster-api-agent/util/log"
 	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
 
 	"github.com/pkg/errors"
 
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type InfraEnvControllerConfig struct {
+	// UseInternalImageURL when set to false means that we'll use the InfraEnv's iso download URL
+	// as is. When set to true, it'll use the assisted-image-service's internal IP as part of the
+	// download URL.
+	UseInternalImageURL bool `envconfig:"USE_INTERNAL_IMAGE_URL" default:"false"`
+	// ImageServiceName is the Service CR name for the assisted-image-service
+	ImageServiceName string `envconfig:"IMAGE_SERVICE_NAME" default:"assisted-image-service"`
+	// ImageServiceNamespace is the namespace that the Service CR for the assisted-image-service is in
+	ImageServiceNamespace string `envconfig:"IMAGE_SERVICE_NAMESPACE" default:""`
```

Comment:
Super minor but we probably don't need the `default` at all here, right?

Reply:
Oh yep good point! Removing it